src/integrationtests/java/com/aws/greengrass/integrationtests/deviceauth/EvaluateClientDeviceActionsWithPolicyVariablesTest.java

```diff
@@ -0,0 +1,267 @@
+        this.certificate = cert;
+        this.clientPem = clientPem;
+    }
+
+    private String getClientDeviceSessionAuthToken(String thingName, String clientPem) throws Exception {
+
+        // create thing - needed for api call to validate thing + certificate
+        ThingRegistry thingRegistry = kernel.getContext().get(ThingRegistry.class);
+        Thing MyThing = thingRegistry.createThing(thingName);
+        MyThing.attachCertificate(certificate.getCertificateId());
+        thingRegistry.updateThing(MyThing);
+
+        // get token for client device to make ipc call
+        ClientDevicesAuthServiceApi api = kernel.getContext().get(ClientDevicesAuthServiceApi.class);
+
+        // create client device session and get token
+        return api.getClientDeviceAuthToken("mqtt", new HashMap<String, String>() {{
+            put("clientId", thingName);
+            put("certificatePem", clientPem);
+            put("username", "foo");
+            put("password", "bar");
+        }});
+    }
+
+    private void startNucleusWithConfig(String configFileName) throws InterruptedException {
+        startNucleusWithConfig(configFileName, State.RUNNING);
+    }
+
+    private void startNucleusWithConfig(String configFileName, State expectedServiceState) throws InterruptedException {
+        CountDownLatch authServiceRunning = new CountDownLatch(1);
+        kernel.parseArgs("-r", rootDir.toAbsolutePath().toString(), "-i",
+                getClass().getResource(configFileName).toString());
+        GlobalStateChangeListener listener = (GreengrassService service, State was, State newState) -> {
+            if (service.getName().equals(ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME)
+                    && service.getState().equals(expectedServiceState)) {
+                authServiceRunning.countDown();
+            }
+        };
+        kernel.getContext().addGlobalStateChangeListener(listener);
+        kernel.launch();
+        assertThat(authServiceRunning.await(30L, TimeUnit.SECONDS), is(true));
+        kernel.getContext().removeGlobalStateChangeListener(listener);
+    }
+
+    private static void authzClientDeviceAction(GreengrassCoreIPCClient ipcClient,
+                                                AuthorizeClientDeviceActionRequest request,
+                                                Consumer<AuthorizeClientDeviceActionResponse> consumer)
+            throws Exception {
+        AuthorizeClientDeviceActionResponseHandler handler =
+                ipcClient.authorizeClientDeviceAction(request, Optional.empty());
+        AuthorizeClientDeviceActionResponse response = handler.getResponse().get(1, TimeUnit.SECONDS);
+        consumer.accept(response);
+    }
+
+    @AfterEach
+    void afterEach() {
+        LogConfig.getRootLogConfig().reset();
+        kernel.shutdown();
+    }
+
+    @Test
+    void GIVEN_cdaWithDeviceGroupPolicy_WHEN_AuthorizeClientDeviceValidResourceAction_THEN_returnTrue() throws Exception {
```

Comment:
naming:  for all the tests, instead of returnTrue/returnFalse, say whether client is authorized or not, since that's what we're testing

Reply:
`GIVEN_permissive_group_policy_with_thing_name_WHEN_client_authorizes_with_thing_name_resource_THEN_client_authorized()`, something along these lines